map-sdk/data-example/src/App.tsx

```diff
@@ -0,0 +1,194 @@
+import {FC, useCallback, useEffect, useRef, useState} from 'react';
+import {createMap, MapApi} from '@unfolded/map-sdk';
+import californiaCities from './data/california-cities.json';
+import arizonaCities from './data/arizona-cities.json';
+
+export const App: FC = () => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [map, setMap] = useState<MapApi | null>(null);
+
+  useEffect(() => {
+    const initMap = async () => {
+      const map = await createMap({
+        container: containerRef.current!
+      });
+
+      setMap(map);
+
+      // here you add map.thing()
+    };
+
+    initMap();
+  }, []);
+
+  return <div id="map-container" ref={containerRef}></div>;
+};
+
+export const AppWithCustomUI: FC = () => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [map, setMap] = useState<MapApi | null>(null);
+  const [datasetAdded, setDatasetAdded] = useState(false);
+  const [jsonPopupVisible, setJsonPopupVisible] = useState(false);
+  const [datasetData, setDatasetData] = useState<any>(null);
+
+  useEffect(() => {
+    const initMap = async () => {
+      const map = await createMap({
+        container: containerRef.current!
+      });
+
+      setMap(map);
+    };
+
+    initMap();
+  }, []);
+
+  // Function to get the dataset ID
+  // (in this example it is always the first/only dataset on the map)
+  const getDatasetId = (): string | null => {
+    if (!map) {
+      return null;
+    }
+
+    const datasets = map.getDatasets();
+
+    if (datasets.length === 1) {
+      return datasets[0].id;
+    } else {
+      console.log('There are no datasets or multiple datasets on the map.');
+      return null;
+    }
+  };
+
+  // Function to handle dataset update logic
+  const updateDataset = (datasetId: string, updateFunction: (dataset: any) => void) => {
+    if (!map) {
+      return;
+    }
+
+    const datasetToUpdate = map.getDatasetById(datasetId);
+
+    if (datasetToUpdate) {
+      updateFunction(datasetToUpdate);
+      map.updateDataset(datasetId, datasetToUpdate);
+    }
+  };
+
+  // Add a dataset to the map
+  const handleAddDataset = useCallback(() => {
+    if (!map || datasetAdded) {
+      return;
+    }
+
+    setDatasetAdded(true);
+    console.log('Dataset added.');
+
+    map.addDataset({
+      id: 'california-cities',
+      label: 'California Cities',
+      data: californiaCities
+    });
+  }, [datasetAdded, map]);
+
+  // Update a dataset's studio metadata, including its label and color
+  // (viewable in Studio's sidebar)
```

Comment:
Pointing out the difference between `updateDataset` and `replaceDataset` seems important as they can accept many of the same properties. (maybe this is something that we should fix on our end)

Reply:
Noted! For now and for this example, it should be clear that one manages the dataset's metadata while the other manages the dataset's data.